python_modules/dagster/dagster_tests/storage_tests/test_local_instance.py

```diff
@@ -284,16 +286,31 @@ def _instantiate_ephemeral_instance(_):
         gc.enable()
 
 
+def test_threadsafe_ephemeral_instance():
+    n = 25
+    with DagsterInstance.ephemeral() as shared:
+
+        def _run(_):
+            shared.root_directory  # noqa: B018
+            with DagsterInstance.ephemeral() as instance:
+                instance.root_directory  # noqa: B018
+            return True
+
+        with ThreadPoolExecutor(max_workers=n) as executor:
+            results = executor.map(_run, range(n))
+            assert all(results)
```

Comment:
? bad rebase? don't add this back

Reply:
yeah must've been a mistake in rebase conflict resolution